pyterrier/index.py

```diff
@@ -247,12 +247,119 @@ def createAsList(files_path : Union[str, List[str]]):
 # Using enum class create enumerations
 class IndexingType(enum.Enum):
     """
-        This enum is used to determine the type of index built by Terrier. The default is CLASSIC.
+        This enum is used to determine the type of index built by Terrier. The default is CLASSIC. For more information,
+        see the relevant Terrier `indexer <https://terrier-core.readthedocs.io/en/latest/indexer_details.html>`_
+        and `realtime <https://terrier-core.readthedocs.io/en/latest/realtime_indices.html>`_ documentation.
     """
     CLASSIC = 1 #: A classical indexing regime, which also creates a direct index structure, useful for query expansion
     SINGLEPASS = 2 #: A single-pass indexing regime, which builds an inverted index directly. No direct index structure is created. Typically is faster than classical indexing.
-    MEMORY = 3 #: An in-memory index. No direct index is created.
+    MEMORY = 3 #: An in-memory index. No persistent index is created.
 
+class TerrierStemmer(Enum):
+    """
+        This enum provides an API for the stemmers available in Terrier. The stemming configuration is saved in the index
+        and loaded at retrieval time. `Snowball <https://snowballstem.org/>`_ stemmers for various languages 
+        `are available in Terrier <http://terrier.org/docs/current/javadoc/org/terrier/terms/package-summary.html>`_.
+    """
+    none = 'none' #: Apply no stemming
+    porter = 'porter' #: Apply Porter's English stemmer
+    weakporter = 'weak' #: Apply a weak version of Porter's English stemmer
```

Comment:
Weak Porter is the only "weak" stemmer of there could be something like "WeakKrovetz"? In this case maybe using 'weakporter' is preferable...